x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,574 @@
+xpack.security.transport.ssl.truststore.path: config/elastic-stack-ca.p12
+xpack.security.transport.ssl.truststore.type: PKCS12
+----
+
+. On every node in your cluster, generate a certificate and private key for your
+node. You must include the CA that was used to sign the certificate currently
+in use.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12
+----
++
+[%collapsible%open]
+.Command parameters
+====
+   `--ca <ca_file>`:: Name of the CA keystore used to sign your certificates. The
+   default file name from the `elasticsearch-certutil` tool is
+   `elastic-stack-ca.p12`.
+====
++
+    a. Enter the password for your CA keystore when prompted.
+
+    b. Enter the name for your keystore, which must match the name of the
+    keystore currently in use.
++
+NOTE: If your 
+<<cert-password-updates,keystore password is changing>>, then save the
+keystore with a new filename. 
+
+    c. (Optional) Enter a password for your keystore. 
+
+. If you entered a password when creating the node certificate, run the
+following command to store the password in the {es} keystore:
++
+--
+[source,shell]
+----
+./bin/elasticsearch-keystore add xpack.security.transport.ssl.keystore.secure_password
+----
+--
+
+. Stop {es} on the first node in your cluster where you are updating the
+keystore.
+
+. Replace your existing keystore with the new keystore, ensuring that the
+file names match. For example, `elastic-stack-ca.p12`.
+
+. If your keystore is password protected, the password is stored in the {es}
+keystore, _and_ the password needs to change, then update the
+`ES_PATH_CONF/elasticsearch.yml` file to use the filename of the new keystore
+file that you created.
+
+. Restart the node where you updated the keystore. 
+
+. (Optional) Use the <<security-api-ssl,SSL certificate API>> to verify that
+{es} loaded the new keystore.
+
+. On each additional node in your cluster, stop {es}, update the keystore,
+and restart {es}.
+
+[[node-certs-same-http]]
+==== Generate a new certificate for the HTTP layer
+The certificate for the HTTP layer is used to verify other components when they
+connect to {es}, such as {kib} or any of the Elastic language clients.
+
+. Stop {es} and {kib} on the first node in your cluster where you are updating
+the HTTP certificate. 
+
+. From the directory where you installed {es}, run the {es} HTTP certificate 
+tool.
++
+[source,shell]
+----
+./bin/elasticsearch-certutil http
+----
++
+This command generates a `.zip` file that contains certificates and keys
+to use with {es} and {kib}. Each folder contains a `README.txt`
+explaining how to use these files.
+
+   a. When asked if you want to generate a CSR, enter `n`.
+
+   b. When asked if you want to use an existing CA, enter `y`.
+
+   c. Enter the absolute path to your CA, such as the path to the
+    `elastic-stack-ca.p12` file.
+
+   d. Enter the password for your CA.
+
+   e. Enter an expiration value for your certificate. You can enter the
```

Comment:
lets add a more safe example, users often confuse our examples for suggestions, let's do 1y

Reply:
++ I'll make that change.